src/bin/pgcopydb/table-data.c

```diff
@@ -382,7 +382,7 @@ copydb_process_table_data_worker(CopyDataSpec *specs)
 		CopyTableDataSpec *tableSpecs = &(tableSpecsArray->array[tableIndex]);
 
 		/* reuse the same connection to the source database */
-		tableSpecs->sourceSnapshot = specs->sourceSnapshot;
+		tableSpecs->sourceSnapshot = &(specs->sourceSnapshot);
```

Comment:
@dimitri if converting the `CopyTableDataSpec->sourceSnapshot` to a pointer seems correct, then we might want to remove this line.